validator.go

```diff
@@ -129,13 +128,16 @@ func (t *TableValidator) Init() error {
 	// Set of fields by position mapped to schema.
 	t.fields = fields
 
-	if len(extra) > 0 || len(missing) > 0 {
+	err_context["unknown_fields"] = unknown
+	err_context["missing_fields"] = missing
+	if len(unknown) > 0 || len(missing) > 0 {
+		match_err = true
+	}
+
+	if length_err || match_err {
 		return &ValidationError{
-			Err: ErrExtraColumns,
-			Context: Context{
-				"extra":   extra,
-				"missing": missing,
-			},
+			Err:     ErrBadHeader,
+			Context: err_context,
```

Comment:
Since the context is going to have all four keys set, then there is no reason to incrementally assign the keys above. Just define the context here and set the fields like it was before.